core/creator/src/main/java/io/quarkus/creator/phase/curate/DefaultUpdateDiscovery.java

```diff
@@ -116,4 +116,55 @@ private String resolveUpToVersion(AppArtifact artifact) throws AppCreatorExcepti
         }
         return majorStr + "." + String.valueOf(minor + 1) + ".alpha";
     }
+
+    private String getFromVersion(AppArtifact artifact) throws AppCreatorException {
+        // here we are looking for the major version which is going to be used
+        // as the base for the version range to look for the updates
+        final String version = artifact.getVersion();
+        final int majorMinorSep = version.indexOf('.');
+        if (majorMinorSep <= 0) {
+            throw new AppCreatorException("Failed to determine the major version in " + version);
+        }
+        final String majorStr = version.substring(0, majorMinorSep);
+        if (updateNumber == VersionUpdateNumber.MAJOR) {
+            final long major;
+            try {
+                major = Long.parseLong(majorStr);
+            } catch (NumberFormatException e) {
+                throw new AppCreatorException(
+                        "The version is expected to start with a number indicating the major version: " + version);
+            }
+            return String.valueOf(major + 1) + ".alpha";
```

Comment:
One small question here: why the `.alpha` part?

Reply:
The short answer would be to include alpha, beta, CR, etc in the range. We could have (and certainly will at some point) a discussion about that. At this point though, this primitive mechanism does not allow you to express the qualifier of the release you are interested in. So even if I remove `.alpha` from that line, alpha releases would still be included in the range starting from `(major + 1).final`.
In perspective, the default lowest acceptable qualifier is going to `final` (an absence of a qualifier means `final`, from that POV all Quarkus releases so far have been `final` releases), with the possibility to explicitly ask for the latest alpha, beta, cr, etc.

Reply:
OK, thanks for the explanation. Maybe we need an issue explaining all that so that we keep that in mind?

Reply:
It would have to a part of the whole `provisioning roadmap`, which I don't think is detailed anywhere at this point. It will have to emerge from the ecosystem discussion/roadmap.
This current mechanism is kind of a playground. In perspective I see it being BOM-based instead of 'per-extension'.

Reply:
OK. Makes sense. Thanks.